crates/energy/src/battery/component.rs

```diff
@@ -0,0 +1,112 @@
+use bevy::prelude::*;
+
+use crate::battery::{MIN_ATTACK_ENERGY, MIN_MOVE_ENERGY};
+
+/// The battery component is used to store the energy level of an entity.
+///
+/// # fields
+/// - `capacity`: The maximum capacity of the battery in kilojoules.
+/// - `energy`: The current energy level of the battery in kilojoules.
+#[derive(Component, Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Battery {
+    capacity: f32, // in kilojoules
```

Comment:
Maybe `capacity_joules` would be a less error prone name? What do you think?

Reply:
How so?

Reply:
Because it include units in the name. The downside is that it is longer. Also, we might start to enforce units at compile time once we get a decent UOM support. Therefore, I do not have a strong opinion on this.

Reply:
Maybe to make it more readable I make an enum that takes different joule units as an input to the battery methods?